arbitrum-docs/launch-orbit-chain/how-tos/customize-arbos.mdx

```diff
@@ -0,0 +1,148 @@
+Add your expected ArbOS version to the switch case statement of nextArbosVersion. Here we will take ArbOS V21 as an example:
+```go
+ensure := func(err error) {
+    if err != nil {
+        message := fmt.Sprintf(
+            "Failed to upgrade ArbOS version %v to version %v: %v",
+            state.arbosVersion, state.arbosVersion+1, err,
+        )
+        panic(message)
+    }
+}
+nextArbosVersion := state.arbosVersion + 1
+switch nextArbosVersion {
+    case 1:
+        //.....
+    case 2:
+        //....
+    //.....
+    case 21:
+        // Set your new ArbOS state value here
+        ensure(state.SetNewMyNumber(${random number}))
+    //....
+} 
+```
+
+Here, we will ensure that the initial value of $(a random number) after ArbOS is upgraded to V21.
+
+
+:::caution
+It should be noted that when you initialize the state (initial code is in [customize precompile Option 5](./customize-precompile.mdx#option-5-call-and-modify-state)), you need to initialize it to 0 or null value first to avoid a potential re-org on your chain.
+:::
+
+Also, please make sure your program cannot call the `state.SetNewMyNumber` or other functions may change the value of `myNumber` before ArbOS V21. 
+To prevent this, if you are using an external call to the precompile contract to change the value, you can refer to [point 1](./customize-arbos.mdx#1-add-a-new-method-to-exsiting-precompile-on-a-specific-arbos-version) or
+[point 2](./customize-arbos.mdx#2-create-a-new-precompile-contract-on-a-specific-arbos-version) to set the activation time of the precompile contract method.
+If your nitro code needs to call this method to change the state, you can continue reading [point 4](./customize-arbos.mdx#4-any-changes-in-the-stf-logic-that-will-affect-the-final-execution-result).
+
+### 4. Any changes in the STF logic that will affect the final execution result
+If you change the logic in STF, it will cause the execution result of the transaction to be different, 
+you need to keep the original execution logic and put the new logic into another branch.
+You can use an `if else` statement to control it.
+
+For example, we change the `SayHi` return after upgrading the ArbOS version:
+```go
+// The method in ArbHi precompile
+func (con *ArbHi) SayHi(c ctx, evm mech) (string, error) {
+    if p.state.ArbOSVersion() >= ${The arbos you want to upgrade to} {
+        // Your new logic code
+        return "hi, new ArbOS version", nil
+    } else {
+        // The old logic code needs to be kept
+        return "hi", nil
+    }
+}
+```
+
+In the above code, we use precompiles as an example. Some logic might also affect the STF, such as the methods in [block_process.go][block_processor_file_link], [internal_tx.go][internal_tx_file_link], [tx_processor.go][tx_processor_file_link] and so on.
+You also need to use those ArbOS control methods for different logic versions.
```

Comment:
Are you referring to the `.go` methods? 
```suggestion
The aforementioned ArbOS control methods will be needed for interacting with different versions of the STF logic.
```

Does this match what you're trying to say?